src/test/java/org/opensearch/ad/bwc/ADBackwardsCompatibilityIT.java

```diff
@@ -83,25 +132,203 @@ public void testBackwardsCompatibility() throws Exception {
             Set<Object> pluginNames = plugins.stream().map(map -> map.get("name")).collect(Collectors.toSet());
             switch (CLUSTER_TYPE) {
                 case OLD:
+                    // Ingest test data
+                    ingestTestDataForHistoricalAnalysis(
+                        client(),
+                        dataIndexName,
+                        detectionIntervalInMinutes,
+                        true,
+                        10,
+                        totalDocsPerCategory,
+                        categoryFieldSize
+                    );
+                    assertEquals(totalDocsPerCategory * categoryFieldSize * 2, getDocCountOfIndex(client(), dataIndexName));
                     Assert.assertTrue(pluginNames.contains("opendistro-anomaly-detection"));
                     Assert.assertTrue(pluginNames.contains("opendistro-job-scheduler"));
-                    createBasicAnomalyDetector();
+
+                    // Create single entity detector and start realtime job
+                    createRealtimeAnomalyDetectorsAndStart(SINGLE_ENTITY_DETECTOR);
+
+                    // Create single category HC detector and start realtime job
+                    createRealtimeAnomalyDetectorsAndStart(SINGLE_CATEGORY_HC_DETECTOR);
+
+                    // Create single entity historical detector and start historical analysis
+                    createHistoricalAnomalyDetectorsAndStart();
+
+                    // Verify cluster has correct number of detectors now
+                    verifyAnomalyDetectorCount(TestHelpers.LEGACY_OPENDISTRO_AD_BASE_DETECTORS_URI, 3);
+                    // Verify cluster has correct number of detectors with search detector API
+                    assertEquals(3, countDetectors(client(), null));
+
+                    // Verify all realtime detectors are running realtime job
+                    verifyAllRealtimeJobsRunning();
                     break;
                 case MIXED:
+                    // We have no way to specify whether send request to old node or new node now.
```

Comment:
Is this comment relevant since you added tests based on rounds?

Reply:
Oh, this should be inside the TODO next line, will move to TODO